### PR TITLE
add nagios checks for digger

### PR DESCRIPTION
**Pull request by laurafitzgerald**
**JIRA**
https://issues.jboss.org/browse/RHMAP-16489

**Changes**
- define hostgroup digger
- define service jenkins
- define command check_jenkins
- Add digger to Memory, Resource etc. checks

**Verification**
Docker image build and pushed to docker hub tag: digger-test
This image is deployed to OSM4 digger project nagios pod

### CODE REVIEW

File: fhservices.cfg.j2 [hunk 1]
```diff
@@ -168,7 +178,7 @@ define service {
        service_description Storage::Pod::Disk Usage
        check_command check_pod_disk_usage!80!90
        use generic-service
-       hostgroup_name core,mbaas
```
Comment (wei-lee) [inline fhservices.cfg.j2]: is it possible to only include this if self-managed build farm is enabled?

File: fhservices.cfg.j2 [hunk 2]
```diff
@@ -184,6 +194,14 @@ define service {
 }
 {% endif %}
 
+define service {
+       service_description jenkins:ping
+       check_command check_jenkins!80!/login
```
Comment (wei-lee) [inline fhservices.cfg.j2]: is this only calling to http port 80? should it be 443?

File: fhservices.cfg.j2 [hunk 3]
```diff
@@ -184,6 +201,33 @@ define service {
...
+       use generic-service
+       hostgroup_name digger
+       contact_groups rhmapadmins
+}
+
+define service {
+       service_description mac_ios:ping
+       check_command check_ios_machines!{{ jenkins_host }}!443!{{ jenkins_user }}!{{ jenkins_pass }}
```
Comment (jessesarn) [inline fhservices.cfg.j2]: The issue with these is that the env variables are pulled into the make script but never pushed into the commands.cfg partial file which eventually gets merged to /etc/nagios/conf.d/fhservices.cfg.

File: fhservices.cfg.j2 (hunk 2 above)
Comment (jessesarn) [inline fhservices.cfg.j2]: 80 is the only port that is listening when tested.  If it should be 443, something else may be wrong as it returns a no route to host when trying to get to jenkins:443

File: fhservices.cfg.j2 (hunk 3 above)
Comment (laurafitzgerald) [inline fhservices.cfg.j2]: @jessesarn good spot. Will pass those now.

**Comment by laurafitzgerald** [comment]
retest this please

File: plugins/default/lib/ios_machine_health.py [hunk 4]
```diff
@@ -0,0 +1,146 @@
...
+
+    unique_statuses = Counter(
+        computer[1]
+        for computer in results
+        )
+
+    ret = max(unique_statuses)
+
+    for computer in results:
+        computer[1]
```
Comment (odra) [inline plugins/default/lib/ios_machine_health.py]: I guess this line could be removed?

Comment (laurafitzgerald) [inline plugins/default/lib/ios_machine_health.py]: yes. agreed :)

**Comment by odra** [comment]
👍 verified both android and ios checks are working

**Comment by laurafitzgerald** [comment]
**Verification**
The image is installer to the [digger project on osm4](https://osm4.skunkhenry.com:8443/console/project/digger/overview)
There are 3 checks to be verified as part of this pr (There is a seperate [JIRA](https://issues.jboss.org/browse/RHMAP-16699) for the failing checks for cpu and memory usage - the problem here is not with the checks)
- android_sdk:Ping
- jenkins:Ping
- mac_os:ping
_Successful checks_
Currently the checks are all coming back successfully in the installed nagios.
_UnSuccessful checks_
- scale down the android-sdk pod
- check should come back CRIT - pod unavailable

**Comment by laurafitzgerald** [comment]
retest this please

**Comment by laurafitzgerald** [comment]
retest this please

**Comment by laurafitzgerald** [comment]
NOTE: Bumped versioon to 4.1.0 through the jenkins build configuration for the pull_request build

**Comment by laurafitzgerald** [comment]
retest this please

**Comment by laurafitzgerald** [comment]
@odra I have added some sections to the README also if you have some time to take a look

**Comment by odra** [comment]
👍  looks good, I think we could improve the nagios scripts by adding some automated tests/create a small base library for check scripts in the future.

**Pull request merged**